packages/engine/src/simulation/engine.rs

```diff
@@ -75,13 +80,18 @@ impl Engine {
         self.run_context_packages(current_step).await?;
         self.run_state_packages().await?;
         let output = self.run_output_packages().await?;
+        let agent_control = if let Some(stop_message) = self.stop.take() {
+            AgentControl::Stop(stop_message)
+        } else {
+            AgentControl::Continue
+        };
         let result = SimulationStepResult {
             sim_id: self.config.sim.id,
             output,
             errors: vec![],
             warnings: vec![],
-            agent_control: AgentControl::Continue, // TODO: OS - Need to pick up from messages
-            stop_signal: false,
+            agent_control,
```

Comment:
Probably fine for this PR, but it feels a bit sloppy to me that we duplicate the stopping signal in `AgentControl::Stop` and `stop_signal`. As in both of those things imply stopping, but we have two because one of them has messages attached.

Unless there are other reasons it seems to me that we should just have an Optional message field within the stop_control. And then the message can be an enum or whatever to say it's from an Agent

Reply:
I also wondered why we even have `stop_signal`. The docs on the field elaborates a bit:
> True if this output signals the stopping of a simulation. Can be False even if a stop signal was sent out before for this simulation run.

However, it's not used anywhere. IMO that's the reason why we have `AgentControl`

Reply:
Yeah I actually think we should remove it for now from `SimulationStepResult` as we always set it to false.

There is an equivalent in `StepUpdate` that is actually used.

Reply:
Removed the field in https://github.com/hashintel/hash/pull/235/commits/0b0d85f381778d1481c5b7e246f12ff35dfec831

Reply:
Basically there were two different types of stop messages -- one about simulation runs that have finished all steps, but haven't persisted their output yet and another one about simulation runs that have fully finished, including persisting output. But we've stopped using the first type of stop message for now.

Reply:
Ah, thank you for the context! :)